code/modules/antagonists/blob/blob_mobs.dm

```diff
@@ -225,24 +225,23 @@
 	if(length(path_list)) //appearantly the solution of using ? infront of the index only works for assoc lists
 		goal_turf = path_list[path_list.len]
 	for(var/w in path_list)
-		if(in_movement && !rally) //incase the spore is already chasing something like a player but the rally command is called
+		if(in_movement > in_movement_current) //incase the spore is already chasing something but something else calls the proc again
```

Comment:
Doesn't this leave in_movement in the state of being incremented forever (since you don't -1 before returns)?

Reply:
in movement will stay like this only if there is another proc of this type on the same mob still running a new proccall will end the old proc like this and so forth once the final proc finishes normally it will be set to 0 again. But in all other cases it needs to stay incemented like this to keep track of new proccals.

Reply:
That way we won't have two different loops trying to move the mob at once i possible different directions

Reply:
I was originally thinking of -1 it but realized there is not much point to it now technically there could be a runtime in this that would keep it incremented but even that wouldn't break it and on the next proccal it would get nulled again in the end.

Reply:
Now you could argue that i should probably call the var differently but i have yet to come up with a better name

Reply:
Trough thinking about it there is one exception where stuff might get scuffed but i have an idea how to fix it only gotta forward the current var as an argument when the proc calls itself else the order might get messed up.